src/io/mod.rs

```diff
@@ -1,5 +1,113 @@
 //! Input and output of images.
+
+use crate::{error, ImageError, ImageResult};
+
 mod reader;
 pub(crate) mod free_functions;
 
 pub use self::reader::Reader;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+/// Set of supported strict limits for a decoder
+pub struct LimitSupport {
```

Comment:
Should there be a 1-1 mapping with fields in the limits struct?

Reply:
The idea was that max_image_width are assumed strict by default and that all decoders support it. As far as I know `ImageDecoder::dimension` is assumed to be correct for all decoders? If so I don't really see any case where one is not able to set strict limits on image size.